packages/web/src/components/premium-content-purchase-modal/PremiumContentPurchaseModal.tsx

```diff
@@ -132,6 +135,11 @@ export const PremiumContentPurchaseModal = () => {
   const { initialValues, validationSchema, onSubmit } =
     usePurchaseContentFormConfiguration({ track })
 
+  const handleClose = useCallback(() => {
+    dispatch(cleanup())
```

Comment:
Would you mind moving this to where we do the other cleanup action (in the form unmount effect above)?
I think there were some potential edge case issues with doing it in response to close here?